test/Rx/Functional/Operator/StartWithTest.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+namespace Rx\Functional\Operator;
+
+use Rx\Functional\FunctionalTestCase;
+use Rx\Observable;
+
+class StartWithTest extends FunctionalTestCase
+{
+
+    /**
+     * @test
+     */
+    public function startWith_never()
+    {
+        $xs = $this->createHotObservable([
+            onNext(150, 1)
+        ]);
+
+        $results = $this->scheduler->startWithCreate(function () use ($xs) {
+            return $xs->startWith(1);
+        });
+
+        $this->assertMessages([
+            onNext(201, 1)
+        ], $results->getMessages());
+    }
+
+    /**
+     * @test
+     */
+    public function startWith_empty()
+    {
+        $xs = $this->createHotObservable([
+            onNext(150, 1),
+            onCompleted(250)
+        ]);
+
+        $results = $this->scheduler->startWithCreate(function () use ($xs) {
+            return $xs->startWith(1);
+        });
+
+        $this->assertMessages([
+            onNext(201, 1),
+            onCompleted(250)
+        ], $results->getMessages());
+    }
+
+    /**
+     * @test
+     */
+    public function startWith_one()
+    {
+        $xs = $this->createHotObservable([
+            onNext(150, 1),
+            onNext(220, 2),
+            onCompleted(250)
+        ]);
+
+        $results = $this->scheduler->startWithCreate(function () use ($xs) {
+            return $xs->startWith(1);
+        });
+
+        $this->assertMessages([
+            onNext(201, 1),
+            onNext(220, 2),
+            onCompleted(250)
+        ], $results->getMessages());
+    }
+
+    /**
+     * @test
+     */
+    public function startWith_multiple()
+    {
+        $xs = $this->createHotObservable([
+            onNext(150, 1),
+            onNext(220, 4),
```

Comment:
@asm89 I'm guessing that  `onNext(202,4)` will never be called, because by the time it's scheduled that time will have passed.

```
  onNext(201, 1),
  onNext(202, 2),
  onNext(203, 3),
  onCompleted(250)
```